internal/log/field/field.go

```diff
@@ -55,7 +55,7 @@ func Coordinate(coordinate coordinate.Coordinate) Field {
 
 // Type builds a Field containing information about a config type. This is used in cases where no full coordinate exists,
 // but only a config type is known - for example in download or deletion
-func Type(t string) Field {
+func Type[X ~string](t X) Field {
```

Comment:
next-level-nit: I'd have preferred a dedicated commit for this change that also tells me WHY this change happened. At a glance/without looking at things like automation.Resource being a dedicated 'string' type elsewhere this looks like a pretty strange change